src/components/basic_table/basic_table.tsx

```diff
@@ -375,12 +379,16 @@ export class OuiBasicTable<T = any> extends Component<
       // used for checking if  initial selection is rendered
       initialSelectionRendered: false,
       selection: [],
+      currentBreakpoint: getBreakpoint(
+        typeof window === 'undefined' ? -Infinity : window.innerWidth
+      ),
```

Comment:
@thanhinhchtom I like the idea here, but I don't actually think this behavior should be dependent on the breakpoint, which is defined by the width of the browser window. Instead, what really matters is the width of the action column - we should only render the action icons on hover if there's room for them to render next to the three-dot-menu icon. 

There is one complication, though, which is the fact that the table becomes responsive at small breakpoints, and the icons no longer appear on hover.

So to summarize, rather than set up a listener on the window size, you should just check the width of the action column (assuming breakpoint > s). And render however many icons (0, 1, or 2) will fit within that column width. I think you still may need a listener for cases where the overall table width changes, because column size can be set as a percentage of the table.

Reply:
@joshuarrrr So I should not use breakpoint here. Instead I should use size of the columnand just fit more button if it is still enough space, right?

Reply:
yes, that's correct. Although you may also still want to have your logic skip small breakpoints, because the different layout there already handles this problem.

Reply:
Okay, I will try that and ask you later if needed?

Reply:
@joshuarrrr One more quick question. I want to clarify the requirement. So I can still use the breakpoint there if needed, but I just cannot hardcode for the different situation as what I did and I have to find the way to use the current size of the component to decide how many buttons will show up, right?

Reply:
The breakpoint usage should just be to determine whether column widths matter at all or not. But yeah the conditional logic for what and how to render should depend on the action column width. And the action column width may be itself dependent on the table/component width.